java/core/src/java/org/apache/orc/impl/InStream.java

```diff
@@ -551,15 +556,20 @@ public void changeIv(Consumer<byte[]> modifier) {
 
     @Override
     public void seek(PositionProvider index) throws IOException {
-      seek(index.getNext());
+      boolean seeked = seek(index.getNext());
       long uncompressedBytes = index.getNext();
-      if (uncompressedBytes != 0) {
-        readHeader();
-        uncompressed.position(uncompressed.position() +
-                              (int) uncompressedBytes);
-      } else if (uncompressed != null) {
-        // mark the uncompressed buffer as done
-        uncompressed.position(uncompressed.limit());
+      if (!seeked && uncompressed != null) {
```

Comment:
Can we get into the case of !seeked && uncompressed == null? I suspect not, but this code would probably be doing the wrong thing by ignoring the case.

Reply:
@omalley !seeked would mean that previously a readHeader has taken place which would result in the uncompressed being allocated, unless when someone seeks to -1. I will add an assert that this is unexpected, if you have any other suggestion please let me know.